api/src/main/java/io/minio/messages/ClientGrantsToken.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.minio.messages;
+
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class ClientGrantsToken {
```

Comment:
@balamurugana is it a domain-specific similarity, or just accidentally similarity? In the first case you absolutely right, and both classes can be merged. In the second case what we will do in case of api changes? Are web identity and client grants requests keep looks the same? It will be difficult to change public api later. So tell me please, is we have a first case or second one.

Reply:
I don't know; and not sure where you got web identity. As both the classes are same except names, you could just do by inheritance or bring an abstract class for both and add additional properties in respective classes.